### PR TITLE
docs: Add contribution guidelines

### PR DESCRIPTION
**Pull request by PIC123**
Add contribution guidelines and templates for pull requests, bug reports, and feature requests for the  plugin repo. Most of the templates and guides are taken from the core Serverless repo to best match the process put forth there, with minor adjustments to reflect the setup of this repo 

AB#805

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
+---
+name: Bug report
+about: Create a report to help us improve
+---
+
+<!--
+1. If you have a question and not a bug report please ask first at http://forum.serverless.com
```
Comment (mydiemho) [inline .github/ISSUE_TEMPLATE/bug_report.md]: even for azure plugin question?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,76 @@
+# Contributing Guidelines
+
+Welcome, and thanks in advance for your help! Please follow these simple guidelines :+1:
+
+# How to contribute to the Serverless Azure Plugin
```
Comment (mydiemho) [inline CONTRIBUTING.md]: also need a section on commit message style guide

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+2. Please check if an issue already exists. This bug may have already been documented
+3. Check out and follow our Guidelines: https://github.com/serverless/serverless/blob/master/CONTRIBUTING.md
```
Comment (pjlittle) [inline .github/ISSUE_TEMPLATE/bug_report.md]: We should update this to point to this repo.

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+- What was the config you used?
+- What stacktrace or error message from your provider did you see?
+
+Similar or dependent issues:
+
+- #12345
+
+## Additional Data
+
+- **_Serverless Framework Version you're using_**:
```
Comment (pjlittle) [inline .github/ISSUE_TEMPLATE/bug_report.md]: we should also add serverless azure functions plugin version.

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+
+```
+npm run lint:fix
+```
+
+To help reduce the effort of creating contributions with this style, an [.editorconfig file](http://editorconfig.org/) is provided that your editor may use to override any conflicting global defaults and automate a subset of the style settings.
+
+# Testing
+
+We aim for a (near) 100% test coverage, so make sure your tests cover as much of your code as possible.
```
Comment (pjlittle) [inline CONTRIBUTING.md]: 80%? Targeting 100 seems unrealistic.

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+The easiest thing you can do to help us move forward and make an impact on our progress is to simply provide support to other people having difficulties with their Serverless projects.
+
+You can do that by replying to [issues on Github](https://github.com/serverless/serverless-azure-functions/issues), chatting with other community members in [our Chat](http://chat.serverless.com) or helping with questions in [our Forum](http://forum.serverless.com).
+
+
+---
+
+# Code Style
+
+We aim for clean, consistent code style. We're using ESlint to check for codestyle issues using the Airbnb preset.
```
Comment (pjlittle) [inline CONTRIBUTING.md]: We can remove the reference to Airbnb; we're using eslint typescript recommended settings.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 7]
```diff
@@ -0,0 +1,51 @@
+<!--
+1. Please check out and follow our Contributing Guidelines: https://github.com/serverless/serverless/blob/master/CONTRIBUTING.md
```
Comment (pjlittle) [inline .github/PULL_REQUEST_TEMPLATE.md]: ditto here.

File: .github/ISSUE_TEMPLATE/feature_request.md [hunk 8]
```diff
@@ -0,0 +1,24 @@
+---
+name: Feature request
+about: Suggest an idea for serverless framework
+---
+
+<!--
+1. If you have a question and not a feature request please ask first at http://forum.serverless.com
+2. Please check if an issue already exists. This feature may have already been requested
+3. Check out and follow our Guidelines: https://github.com/serverless/serverless/blob/master/CONTRIBUTING.md
```
Comment (pjlittle) [inline .github/ISSUE_TEMPLATE/feature_request.md]: update this url reference as well

**Pull request merged**